src/lib/spriteActor.js

```diff
@@ -0,0 +1,28 @@
+async function loadImage(src) {
+    let img = new Image();
+    img.src = src;
+    return new Promise((res, rej) => {
+        img.onload = () => {
+            res(img);
+        }
+    });
+}
+
+export default class SpriteActor extends Actor{
+    constructor(bounds, properties, states){
+        super(bounds);
+        this.properties = properties || {};
+        this.states = states || {idle: {location: null, rps: 5, loop: true, frames: [{x: 0, y: 0, width: 0, height: 0}]}};
```

Comment:
Might want to create a built-in function to load states and properties from a spriteObj. We can probably get the properties from the schema.

```
loadSpriteObj = (spriteObj) => {
   //parse properties
   
   //parse states

   //calculate diffing images and load them
}
```